rs/rosetta-api/icp_ledger/rosetta-integration-tests/tests/tests.rs

```diff
@@ -1327,3 +1328,142 @@ async fn test_network_status_single_genesis_transaction() {
+            None,
+        )
+        .await
+        .unwrap();
+
+    // Let's create a few transactions to make sure rosetta is working with the mainnet ledger version
+    for _ in 0..100 {
+        let transfer_arg = TransferArg {
+            from_subaccount: None,
+            to: Account::from(Principal::anonymous()),
+            fee: None,
+            created_at_time: None,
+            memo: None,
+            amount: Nat::from(1u64),
+        };
+        let arg = Encode!(&transfer_arg).unwrap();
+        env.pocket_ic
+            .update_call(env.ledger_id, env.sender_id, "icrc1_transfer", arg)
+            .await
+            .unwrap();
+    }
+
+    // We should now have 100 blocks
+    // Let's restart rosetta to make sure it can handle the mainnet ledger version
+    env.restart_rosetta_node(false, false).await.unwrap();
+    env.rosetta.wait_until_synced_up_to(100).await.unwrap();
+
+    // Let's check the network status
+    let network_status = env.rosetta.network_status().await.unwrap();
+    let current_block = env
+        .rosetta
+        .block(PartialBlockIdentifier {
+            index: Some(100),
+            hash: None,
+        })
+        .await
+        .unwrap()
+        .block
+        .unwrap();
+    let genesis_block = env
+        .rosetta
+        .block(PartialBlockIdentifier {
+            index: Some(0),
+            hash: None,
+        })
+        .await
+        .unwrap()
+        .block
+        .unwrap();
+    assert_eq!(
+        network_status.current_block_identifier,
+        current_block.block_identifier
+    );
+    assert_eq!(
+        network_status.genesis_block_identifier,
+        genesis_block.block_identifier
+    );
+
+    // Now we upgrade the ledger canister to the latest version of the current branch
+    let ledger_wasm_current_branch = icp_ledger_wasm_bytes();
+    env.pocket_ic
+        .upgrade_canister(
+            env.ledger_id,
+            ledger_wasm_current_branch,
+            Encode!(&LedgerCanisterUpgradePayload(
+                LedgerCanisterPayload::Upgrade(None)
+            ))
+            .unwrap(),
+            None,
+        )
+        .await
+        .unwrap();
+
+    // Let's create a few transactions to make sure rosetta is working with the current branch ledger version
+    for _ in 0..100 {
+        let transfer_arg = TransferArg {
+            from_subaccount: None,
+            to: Account::from(Principal::anonymous()),
+            fee: None,
+            created_at_time: None,
+            memo: None,
+            amount: Nat::from(1u64),
+        };
+        let arg = Encode!(&transfer_arg).unwrap();
+        env.pocket_ic
+            .update_call(env.ledger_id, env.sender_id, "icrc1_transfer", arg)
+            .await
+            .unwrap();
+    }
+
+    // We should now have 200 blocks
+    env.restart_rosetta_node(false, false).await.unwrap();
+    env.rosetta.wait_until_synced_up_to(200).await.unwrap();
```

Comment:
Could we also check the balance of the account receiving the transactions?

Reply:
We already have tests for checking the balances of rosetta versus the icp ledger.